driver-core/src/main/com/mongodb/internal/connection/AbstractMultiServerCluster.java

```diff
@@ -316,8 +318,9 @@ private boolean isStalePrimary(final ServerDescription newDescription) {
             return false;
         }
 
-        return (maxSetVersion.compareTo(newDescription.getSetVersion()) > 0
-                || (maxSetVersion.equals(newDescription.getSetVersion()) && maxElectionId.compareTo(newDescription.getElectionId()) > 0));
+        Integer setVersion = newDescription.getSetVersion();
```

Comment:
Why do we have to make this change for setVersion but not for electionId (used in `compareTo` below)?

Reply:
Because compareTo can take a null value?

Reply:
Weird.  In IntellIJ I see `int compareTo(@NotNull T o)` and `public boolean equals(Object obj)`.  

Anyway, not that important so I'll resolve.

Reply:
Ah I reverted it and then saw it - maxElectionId is an ObjectId so it's compareTo can handle null.